packages/contracts-bedrock/src/interop/L2CrossDomainMessenger.sol

```diff
@@ -0,0 +1,147 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.15;
+
+import { L2CrossDomainMessenger } from "src/L2/L2CrossDomainMessenger.sol";
+
+import { Constants } from "src/libraries/Constants.sol";
+import { Encoding } from "src/libraries/Encoding.sol";
+import { Predeploys } from "src/libraries/Predeploys.sol";
+import { SafeCall } from "src/libraries/SafeCall.sol";
+
+contract InteropEnabledL2CrossDomainMessenger {
+    // new message versionn
+    uint16 public constant INTEROP_MESSAGE_VERSION = 2;
+
+    bytes32 public constant ETH_MAINNET_ID = bytes32(hex"01");
```

Comment:
Can you hardcode the actual `bytes32` value? `constant`s are not actually constants, this will likely emit code for doing type conversions if done this way. also why is it not a `uint256`?

Reply:
using `bytes` since we don't have a definite idea as to what the chain id should be. If we decide on chain id, then `uint256` is the appropriate type.

however depending on the L1 changes, using the system config address or a hash of many fields may be more suitable for the identifier